core.js

```diff
@@ -1140,11 +1140,14 @@ export class FileTypeParser {
 		}
 
 		if (
-			this.check([0x57, 0x45, 0x42, 0x56, 0x54, 0x54, 0x00])		// WEBVTT<EOF>
-			|| this.check([0x57, 0x45, 0x42, 0x56, 0x54, 0x54, 0x0A])	// WEBVTT<LF>
-			|| this.check([0x57, 0x45, 0x42, 0x56, 0x54, 0x54, 0x0D])	// WEBVTT<CR>
-			|| this.check([0x57, 0x45, 0x42, 0x56, 0x54, 0x54, 0x09])	// WEBVTT<Tab>
-			|| this.check([0x57, 0x45, 0x42, 0x56, 0x54, 0x54, 0x20])	// WEBVTT<Space>
+			this.checkString('WEBVTT')
+			&& (
+				this.check([0x00], {offset: 6})		// EOF
+				|| this.check([0x0A], {offset: 6})	// LF
+				|| this.check([0x0D], {offset: 6})	// CR
+				|| this.check([0x09], {offset: 6})	// Tab
+				|| this.check([0x20], {offset: 6})	// Space
+			)
```

Comment:
Sorry that I'm asking so many questions: Should I be checking some of these as strings/escape sequences too since they're all ASCII? Also, is there a way to avoid repeating the offset? I tried doing an `await tokenizer.ignore(6)` after the `WEBVTT` check but that didn't seem to work.

Reply:
If you call `await tokenizer.ignore(6)`, you permanently move the offset with 6 bytes. That also require you need to re-read the buffer used by `this.check`, starting on that offset.

But, that approach is irreversible, meaning all following tests will fail, as they can not ,longer read from offset 0.
Therefor, if you read of (or ignore) from the tokenizer, you have to return the file type or return undefined. You essentially can only do that, if you are sure that remaining tests are pointless to run.

Your code looks fine to me, maybe a bit lengthy as a result of the repeating offset argument, but very readable.

If you prefere you could shorten it by putting the special character in an array, and calling [every] (not tested):

```js
[0x00, 0x0A, 0x0D, 0x09, 0x20].every(lastChar => this.check([lastChar], {offset: 6}))
```

> Should I be checking some of these as strings/escape sequences

I think this is fine without, but you can try, and see if gets more elegant.

Reply:
Thanks for the help!